### PR TITLE
WC analytics customers endpoint

### PR DESCRIPTION
**Pull request by kidinov**
Adds `wc_analytics/reports/customers` end point

Can be tested via tests or the PR - https://github.com/woocommerce/woocommerce-android/pull/9212

### CODE REVIEW

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/customer/CustomerRestClient.kt [hunk 1]
```diff
@@ -107,4 +114,49 @@ class CustomerRestClient @Inject constructor(private val wooNetwork: WooNetwork)
 
         return response.toWooPayload()
     }
+
+    /**
+     * Makes a GET call to `wc-analytics/reports/customers` to fetch customers
+     *
+     */
+    suspend fun fetchCustomersFromAnalytics(
```
Comment (AnirudhBhat) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/customer/CustomerRestClient.kt:122]: NP: ❓ What do you think about writing unit tests for this class?

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/customer/WCCustomerMapper.kt [hunk 2]
```diff
@@ -50,36 +51,71 @@ class WCCustomerMapper @Inject constructor() {
...
+            lastName = model.lastName,
+            username = model.username,
+            billing = billing,
+            shipping = shipping
         )
     }
+
+    @Suppress("ComplexMethod")
+    fun mapToModel(site: SiteModel, dto: CustomerFromAnalyticsDTO): WCCustomerModel {
+        return WCCustomerModel().apply {
```
Comment (AnirudhBhat) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/customer/WCCustomerMapper.kt:89]: NP: ❓ What do you think about writing tests for this mapper? I'm specifically thinking about testing the pieces that involve some kind of business logic like `isPayingCustomer`, `firstName`, and `lastName`.

**Review by AnirudhBhat — APPROVED** [review]
LGTM :shipit: 

Just left a couple of minor comments that are non-blocking.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/customer/CustomerRestClient.kt (hunk 1 above)
Comment (kidinov) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/customer/CustomerRestClient.kt:122]: Added basic test here b379f268dad4976100218436adb051f6b7affb4d

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/customer/WCCustomerMapper.kt (hunk 2 above)
Comment (kidinov) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/customer/WCCustomerMapper.kt:89]: Good point

324aea8486512735ad2deccae352f8a10abf325a

**Pull request merged**